components/service/glean/src/main/java/mozilla/components/service/glean/private/MemoryDistributionMetricType.kt

```diff
@@ -0,0 +1,112 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.service.glean.private
+
+import androidx.annotation.VisibleForTesting
+import mozilla.components.service.glean.Dispatchers
+import mozilla.components.service.glean.histogram.FunctionalHistogram
+import mozilla.components.service.glean.storages.MemoryDistributionsStorageEngine
+import mozilla.components.support.base.log.logger.Logger
+
+/**
+ * This implements the developer facing API for recording memory distribution metrics.
+ *
+ * To prevent the number of buckets from being unbounded, values larger than 1 TB
+ * are truncated to 1 TB.
```

Comment:
No -- desktop has explicitly an set min/max/bucket_count.  Here we're trying to do something "parameterless" so that data is less likely to end up in an under or overflow bucket.

We will need to confirm that 1TB is sufficient -- it is for the histograms on the GeckoView shortlist.